packages/broker/src/plugins/operator/OperatorFleetState.ts

```diff
@@ -41,14 +35,33 @@ export class OperatorFleetState extends EventEmitter<OperatorFleetStateEvents> {
     private readonly ready = new Gate(false)
     private subscription?: Subscription
 
-    constructor(
+    static createOperatorFleetStateBuilder(
+        streamrClient: StreamrClient,
+        heartbeatIntervalInMs: number,
+        pruneAgeInMs: number,
+        pruneIntervalInMs: number,
+        latencyExtraInMs: number,
+        timeProvider = Date.now
+    ): CreateOperatorFleetStateFn {
+        return (coordinationStreamId) => new OperatorFleetState(
```

Comment:
Is this actually a factory? It creates new instances with same config values. In builders the focus is usually to control what kind of configs we apply for each generated object?

Reply:
The concept that varies here is the `coordinationStreamId`. We want to create instances of `OperatorFleetState` for different `coordinationStreamId` with the other config values being held constant.

Reply:
For me factory sounds like a class so I avoided using that name.

Reply:
Yeah, for me it doesn't make a difference. Maybe this use case a kind of hybrid.

Usually it think factory like this:
```
const factory = ...
const instance1 = factory()
const instence2 = factory()
```

And builder something like this:
```
const instance = builder()
    .withFoo(123)
    .withBar(234)
    .build()    
```